### ISSUE TITLE
Compatibilty issue with cheap NRF51822

### ISSUE THREAD

**Issue by waxd86** [virtualabs/btlejack-firmware#17]
Hi virtualabs and everyone !

I saw you have worked on a version of btlejack-firmware for cheap RTL51822 but no news since (https://github.com/virtualabs/btlejack/issues/12#issuecomment-413928888)
If you have some clues and information about it, I'll take it !

PS : I'm actually working to port btlejack-firmware into mbed-os 2 since microbit was impossible to compile (microbit registry is dead).

Thank you so much for ur work!

**Comment by jumpingbrowntrout** [comment]
How did you go with this @waxd86 ? Did you have success?
Did you get an aliexpress rlt51822?
I'm thinking about buying one but don't want to bother if no one has got it working.

**Comment by virtualabs** [comment]
Hi there !

I've spent some time recently working on a nRF52 compatible firmware for the BBC Micro:Bit v2, and the code has been published here https://github.com/virtualabs/btlejack-firmware-nrf52. It has been integrated into btlejack and can be flashed automatically with the `-i` option. This firmware has some known limitations (it hardly detects access addresses from in the air packets) but works fine for sniffing new connections. The code is heavily based on the one used in the first version. So if you own a Micro:Bit v2, it may worth the shot.

Hope this helps :p

**Comment by jumpingbrowntrout** [comment]
Thanks for the update @virtualabs I can see Australia has these cute microbit now apparently. I could go buy one. Half the price of a rf51822.
Just to be clear can these V2 device sniff existing connections for their data?

**Comment by virtualabs** [comment]
TLDR: no, sniffing existing connections is unreliable with this nRF52 compatible firmware.

At the moment sniffing existing connections works if you are lucky enough for the AA to be caught by the firmware (but most of the time it does not, I had many complaints about it in various issues). Nordic's nRF52 series are different from the nRF51 series and therefore sniffing raw packets has became quite difficult with these nRF52 SoCs. 

There are some clues about how to improve this but I need more time to try them and see what works best before releasing a new version.
